OpenSim/Common/Component.cpp

```diff
@@ -134,11 +152,19 @@ void Component::finalizeFromProperties()
 }
 
 // Base class implementation of virtual method.
-// Call extendFinalizeFromProperties on all components
+// Call finalizeFromProperties on all subcomponents
 void Component::componentsFinalizeFromProperties() const
```

Comment:
@aymanhab it is already possible to get useless messages when a component is unnamed. The individual component name does not have to be unique but its path name must be. So unless there is already a duplicate path having a non-unique names should be just fine. For example, all GeometyPaths were unnamed, and there is no reason to name it something <MuscleName>_geometry_path since the path name would be <ModelName>/<MuscleName>/geometry_path anyways. If you have a good scheme for unique naming let me know. Randomly generating a number to tack on or using time will cause de/serialization test failures were we are comparing to an old standard file that has no names, we have to guarantee the same names are autogenerated also when a nameless component is copied.